### PR TITLE
Use SFTPv3Client instead of deprecated SFTPClient

### PR DESCRIPTION
**Pull request by darxriggs**

### CODE REVIEW

File: src/main/java/hudson/plugins/sshslaves/SFTPClient.java [hunk 1]
```diff
@@ -1,150 +0,0 @@
...
-
-import java.io.IOException;
-import java.io.OutputStream;
-import java.io.InputStream;
-
-/**
- * TODO: moved to Jenkins core, so pick it up from there.
- * @author Kohsuke Kawaguchi
- */
-public class SFTPClient extends SFTPv3Client {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SFTPClient.java]: Incompatible

File: src/main/java/hudson/plugins/sshslaves/SFTPFileSystem.java [hunk 2]
```diff
@@ -38,9 +39,9 @@
  * @author Kohsuke Kawaguchi
  */
 class SFTPFileSystem implements FileSystem {
-    private final SFTPClient sftp;
+    private final SFTPv3Client sftp;
 
-    public SFTPFileSystem(SFTPClient sftp) {
+    public SFTPFileSystem(SFTPv3Client sftp) {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SFTPFileSystem.java]: Binary incompatible. Maybe makes sense to create a new SFTPv3FileSystem class instead

**Comment by oleg-nenashev** [comment]
@darxriggs If you have any concerns about y feedback, feel free to respond here

**Comment by darxriggs** [comment]
Thanks for the feedback. I see the point with this PR in not being backward compatible.

I think adding annotations and another class is not worth it when old code cannot be removed anyway. So I just closed the PR.

**Pull request closed without merging**